### DISCUSSION TITLE
Evaluate a pretrained .msgpack model at arbitrary <loc, view_dir> values?

### DISCUSSION THREAD

**Discussion by stanlew7531** [NVlabs/instant-ngp#1294] [Q&A]
Hello!

I've been playing around with this tool as well as the NeRF-SLAM project and have been very impressed! The speed and rendering fidelity out of the GUI have been great!

That being said, one of the things I'm interested in doing is performing meta-analysis between models trained on similar, but not quite identical scenes. I'd like to be able to take in several .msgpack files, and evaluate each of them at arbitrary `<x,y,z,theta,phi>` values, getting the `<r,g,b,sigma>` values in return. I must admit however, that the best way to do this has eluded me. The python bindings appear to expose a render function, but not the underlying implicit model itself. Reading the `testbed_nerf.cu` and other testbed files also doesn't provide me with much insight into the best way to expose this functionality.

Any help would be greatly appreciated!

**Comment by Tom94** [comment]
Hi, you can set `testbed.camera_matrix` via the Python bindings. The expected format is a 3x4 matrix, where the left 3x3 block is the camera rotation and the right 3x1 vector the translation.
